### ISSUE TITLE
`browser.runtime.onMessage.addListener` callback's 3rd argument `sendResponse` has incorrect type

### ISSUE THREAD

**Issue by bnn1** [wxt-dev/wxt#299]
### Describe the bug

The third callback argument `sendResponse` of `browser.runtime.onMessage.addListener` has incorrect signature: 
```ts
sendResponse: () => void
```

### To Reproduce

[Reproduction](https://stackblitz.com/~/github.com/bnn1/wxt-bug-report)

1. Open `entrypoints>background.ts`
2. See TS error "Expected 0 arguments, but got 1.ts(2554)"

OR 

1. add `onMessage` listener: 
```ts
browser.runtime.onMessage.addListener((message, sender, sendResponse) => {
  console.log("Message!:", message)
  sendResponse("Sup brother?")
})
```
2. See TS error "Expected 0 arguments, but got 1.ts(2554)"
3. But response is actually send back

### Expected behavior

`sendMessage` should accept `any`, `unknown` or generic data type
```ts
sendResponse: <T>(data?: T) => void
```

### Screenshots

![image](https://github.com/wxt-dev/wxt/assets/55663639/2d9430c9-c22d-469f-a157-83b939d6a14b)
![image](https://github.com/wxt-dev/wxt/assets/55663639/60579c9d-4b6f-4869-87c8-26275ecd949e)
![image](https://github.com/wxt-dev/wxt/assets/55663639/96a40fef-a354-457a-888a-758e7a530f75)

### Environment

```sh
  System:
    OS: Linux 6.6 EndeavourOS
    CPU: (12) x64 AMD Ryzen 5 4600H with Radeon Graphics
    Memory: 17.76 GB / 30.73 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 18.14.2 - ~/.nvm/versions/node/v18.14.2/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v18.14.2/bin/yarn
    npm: 9.6.0 - ~/.nvm/versions/node/v18.14.2/bin/npm
    pnpm: 8.12.1 - ~/.nvm/versions/node/v18.14.2/bin/pnpm
    bun: 1.0.0 - ~/.bun/bin/bun
  npmPackages:
    wxt: ^0.12.0 => 0.12.5 
```

### Contribution

- [x] I would like to fix this BUG via a PR

**Comment by aklinker1** [comment]
This is likely an upstream issue with `@types/webextension-polyfill`.

Alternatively, you could use the promise API without `sendResponse`. 

![Screenshot_20231225-133710.png](https://github.com/wxt-dev/wxt/assets/10101283/8c9b8e7c-ee63-44b2-b57d-4ed86da28c0b)

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage#examples

```ts
browser.runtime.onMessage.addListener(() => {
  return Promise.resolve("greetings");
});

// or use an async callback

browser.runtime.onMessage.addListener(async () => {
  return "greetings";
});
```

**Comment by dvlden** [comment]
Can someone explain to me why message listener can return something?
On Chrome API I clearly see that it can return only `boolean` or `undefined` but on FF it says return Promise... 

What is the return for?

**Comment by aklinker1** [comment]
@dvlden maybe you want to make a fetch request from a content script, but it's blocked by CORS. To get around this, you can send a message to the background, and perform the request there instead, where CORS is disabled. So the background, the message receiver, needs to be able to send the response back to the content script.

If you need to do async work, like a fetch request, then:

- On Chrome, you `return true` from the message listener and use callbacks to call `sendResponse` once the async work has resolved
- On Firefox, you just return a promise

The polyfill supports both ways of sending a response on both browsers. I think promises are simpler, so I use the Firefox approach in my code.

**Comment by aklinker1** [comment]
Was gonna open a PR to , but this was already taken care of, see https://github.com/Lusito/webextension-polyfill-ts/pull/96. Just hasn't been released to definitely typed yet.

**Comment by andrewshvv** [comment]
> * On Firefox, you just return a promise of the response

Well, if you handle several types of events, and use `async` handler with return or without return it might interfere with the other handlers.

**Comment by aklinker1** [comment]
Just released an experiment feature in `wxt@0.19` to switch to `@types/chrome`, which has this type fixed. Try it out and let me know how it goes: https://github.com/wxt-dev/wxt/issues/868

**Comment by 0x7a7a** [comment]
No more errors in my project, finally! 
Good job
